R/tolerance.cca.R

```diff
@@ -66,16 +67,20 @@ tolerance.cca <- function(x, choices = 1:2,
         if(doN2) {
             y <- sweep(Y, 1, Ytot, "/")^2
             N2 <- 1 / rowSums(y, na.rm = TRUE) ## 1/H
-            res <- sweep(res, 1, sqrt(1 - (1/N2)), "/")
+            ## avoid almost-1 for sites with only one spp
+            N2 <- zapsmall(N2 - 1L) + 1L
```

Comment:
This can be zapped to 1: zapping removes the last (non-significant) bits, but it cannot be zapped to 0 which keeps the first (and only) non-zero bits: So change to

N2 <- zapsmall(N2)